subprojects/gradle/test-project/src/main/kotlin/com/avito/test/gradle/dependencies/GradleDependency.kt

```diff
@@ -0,0 +1,75 @@
+package com.avito.test.gradle.dependencies
+
+import com.avito.test.gradle.dependencies.GradleDependency.Coordinate.Maven
+import com.avito.test.gradle.dependencies.GradleDependency.Coordinate.Platform
+import com.avito.test.gradle.dependencies.GradleDependency.Coordinate.Project
+import org.funktionale.either.Either
+import org.gradle.util.Path
+
+data class GradleDependency(
+    private val configuration: CONFIGURATION,
+    private val dep: GradleDependency.Coordinate
+) : GradleScriptCompatible {
+
+    enum class CONFIGURATION(private val representation: String) : GradleScriptCompatible {
+        COMPILE("compile"),
+        IMPLEMENTATION("implementation"),
+        TEST_IMPLEMENTATION("testImplementation"),
+        ANDROID_TEST_IMPLEMENTATION("androidTestImplementation"),
+        API("api");
+
+        override fun getScriptRepresentation() = representation
+    }
+
+    sealed class Coordinate : GradleScriptCompatible {
+
+        data class Maven(val mavenCoordinates: String) : Coordinate() {
+            override fun getScriptRepresentation() = "'$mavenCoordinates'"
+        }
+
+        data class Project(val projectPath: Path) : Coordinate() {
+            override fun getScriptRepresentation() = "project('${projectPath.path}')"
+        }
+
+        data class Platform(val coordinate: Either<Maven, Project>) : Coordinate() {
+            override fun getScriptRepresentation(): String {
+                return "platform(${
+                    coordinate.fold(
+                        { maven -> maven.getScriptRepresentation() },
+                        { project -> project.getScriptRepresentation() }
+                    )
+                })"
+            }
+        }
+    }
+
+    override fun getScriptRepresentation(): String {
+        return "${configuration.getScriptRepresentation()}(${dep.getScriptRepresentation()})"
+    }
+
+    companion object {
+        fun maven(
+            configuration: CONFIGURATION = CONFIGURATION.IMPLEMENTATION,
+            mavenCoordinate: String
+        ): GradleDependency = GradleDependency(configuration, Maven(mavenCoordinate))
+
+        fun project(
+            configuration: CONFIGURATION = CONFIGURATION.IMPLEMENTATION,
+            projectPath: String
+        ): GradleDependency = GradleDependency(configuration, Project(Path.path(projectPath)))
+
+        fun platformProject(
```

Comment:
```suggestion
        fun platform(
```

The closer to the original API, the better

```
api(platform("com.fasterxml.jackson:jackson-bom:2.9.8"))
api(platform(project(":module")))
```

Reply:
Stay without changes

Reply:
Why?